### DISCUSSION TITLE
Is there a way to go back to the lazy vim dashboard when all buffers are closed

### DISCUSSION THREAD

**Discussion by theesfeld** [LazyVim/LazyVim#3901] [Q&A]
as the title says, when I close all buffers, is there a way to go back to the dashboard automatically?

**Comment by prerit714** [comment]
```:Alpha``` works, try doing that.

**Reply by folke** [reply]
or `:Dashboard`

**Comment by jyuan0** [comment]
Maybe something like this? Adapted from: https://github.com/goolord/alpha-nvim/discussions/85

```lua
vim.api.nvim_create_autocmd("BufDelete", {
  group = vim.api.nvim_create_augroup("dashboard_on_empty"),
  callback = function(args)
    local deleted_name = vim.api.nvim_buf_get_name(args.buf)
    local deleted_ft = vim.api.nvim_get_option_value("filetype", { buf = args.buf })
    local dashboard_on_empty = (deleted_name == "" and deleted_ft == "")
      or (vim.api.nvim_buf_get_name(0) == "" and vim.api.nvim_get_option_value("filetype", { buf = 0 }) == "")
    if dashboard_on_empty then
      vim.cmd("Dashboard")
    end
  end,
})
```

Update: this code is buggy and misses lots of edge cases.

**Reply by RHOOPH** [reply]
Thank you . 

I feel like this should have been the default behaviour

**Reply by thesobercoder** [reply]
Any reason why is this not the default behavior?

**Reply by dpetka2001** [reply]
Because not all ppl want this behavior? And those who want it, can add it themselves.

**Reply by thesobercoder** [reply]
Can be an optional default for those who want it, no? Meaning that it's there but turned off by default.

**Reply by dpetka2001** [reply]
It's an autocmd that will be executed. You either put this autocmd in your `autocmds.lua` file to execute it or you don't put it at all. You can hook it up upon a global variable, that the user should have to enable in his `options.lua` file, but in my personal opinion this sets precedence for many other personal preferences, that it's simply not possible to take every single user's personal preferences into account. 

And in the end this is the whole reason of the user personal configuration. For the users to customize things according to their own personal preferences. That is my humble opinion as just a simple user like you.

**Reply by thesobercoder** [reply]
I agree with you, but people who are not proficient with Lua like me (and new to Neovim), will probably have a hard time figuring it out, unless they accidentally happen to land on this discussion. Maybe some documentation will suffice as well.

**Reply by dpetka2001** [reply]
I'm just a simple user like you with no access and no say in this project (I know I've already mentioned it in my previous comment, but I just want to make sure that there are no misunderstandings). Everything I expressed was my personal opinion. 

You can make your case to the maintainer and even submit a PR for improving the documentation in the [documentation repo here](https://github.com/LazyVim/lazyvim.github.io). I guess it could be included in the `Recipes` section, but it's not my call. Feel free to submit a PR with this autocmd under the `Recipes` section and the maintainer will decide.

**Reply by folke** [reply]
good idea to add it to recipes if anyone wants to create a PR

**Reply by iamxiaojianzheng** [reply]
> good idea to add it to recipes if anyone wants to create a PR

https://github.com/LazyVim/lazyvim.github.io/pull/101

**Reply by jyuan0** [reply]
@RHOOPH, @thesobercoder, @iamxiaojianzheng - just so you know, the code is buggy. Depending on how you use neovim, you might not run into the bugs, but if you all of sudden see the dashboard when you didn't expect it, this autocommand is probably the cause :smiley:

**Reply by thesobercoder** [reply]
@jyuan0 I turned it off for now. The specter plugin doesn't work with this turned on.

**Comment by dpetka2001** [comment]
So, based on @jyuan0's linked post on `alpha`'s repo, I saw that people over there were using a plugin [bufdelete.nvim](https://github.com/famiu/bufdelete.nvim) which provides a `User` event `BDeletePost`. I tried to replicate that behavior with the following
```lua
vim.api.nvim_create_autocmd("BufDelete", {
  group = vim.api.nvim_create_augroup("bufdelpost_autocmd", {}),
  desc = "BufDeletePost User autocmd",
  callback = function()
    vim.schedule(function()
      vim.api.nvim_exec_autocmds("User", {
        pattern = "BufDeletePost",
      })
    end)
  end,
})

vim.api.nvim_create_autocmd("User", {
  pattern = "BufDeletePost",
  group = vim.api.nvim_create_augroup("dashboard_delete_buffers", {}),
  desc = "Open Dashboard when no available buffers",
  callback = function(ev)
    local deleted_name = vim.api.nvim_buf_get_name(ev.buf)
    local deleted_ft = vim.api.nvim_get_option_value("filetype", { buf = ev.buf })
    local deleted_bt = vim.api.nvim_get_option_value("buftype", { buf = ev.buf })
    local dashboard_on_empty = deleted_name == "" and deleted_ft == "" and deleted_bt == ""

    if dashboard_on_empty then
      vim.cmd("Dashboard")
    end
  end,
})
```
There might be other additional checks that would have to be added for other edge cases that I failed to think of, because I just tried it out for a really short time and am not particularly interested in using this. With the above, the problem when you create an empty buffer with `<leader>fn` and then rename the buffer and the Dashboard appears, does not happen any more from my short testing (because the `User BufDeletePost` event is scheduled to happen after the `BufDelete` event which happens before you delete a buffer from the buffer list or you rename a buffer, so all actions are taken after `BufDelete`).

If people want to try this out, do bear in mind that this is not well tested and you probably would have to make additional adjustments for edge cases that I didn't take into consideration, but I believe it's closer to what people on the Alpha repo thread were using. You might as well just use the plugin `bufdelete.nvim`, which I assume is more well tested and take advantage of the features it provides to do what you want, but I haven't used that either, so I can't answer any questions regarding its usage.

**Reply by jyuan0** [reply]
Nice! Thanks for taking the time to look!

**Reply by Sparrow1029** [reply]
Just an update: This does work for me (so far), but I changed the `"Dashboard"` command to support the default `Snacks.nvim` plugin now

```lua
-- ...
    if dashboard_on_empty then
      Snacks.dashboard.open()
    end
  end,
})
```
